csunplugged/tests/topics/loaders/test_lessons_loader.py

```diff
@@ -0,0 +1,457 @@
+                    'level': 2,
+                    'text': 'Heading 2',
+                    'slug': 'heading-2',
+                    'children': [
+                        {
+                            'level': 3,
+                            'text': 'Heading 3',
+                            'slug': 'heading-3',
+                            'children': [],
+                        }
+                    ],
+                },
+                {
+                    'level': 2,
+                    'text': 'Heading 4',
+                    'slug': 'heading-4',
+                    'children': [],
+                }
+            ],
+        )
+
+    def test_missing_min_age_value(self):
+        config_file = os.path.join(self.loader_name, "missing-min-age.yaml")
+        complete_config_file_path = os.path.join(self.test_data.LOADER_ASSET_PATH, config_file)
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        self.assertRaises(
+            MissingRequiredFieldError,
+            lesson_loader.load,
+        )
+
+    def test_missing_max_age_value(self):
+        config_file = os.path.join(self.loader_name, "missing-max-age.yaml")
+        complete_config_file_path = os.path.join(self.test_data.LOADER_ASSET_PATH, config_file)
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        self.assertRaises(
+            MissingRequiredFieldError,
+            lesson_loader.load,
+        )
+
+    def test_missing_number_value(self):
+        config_file = os.path.join(self.loader_name, "missing-number.yaml")
+        complete_config_file_path = os.path.join(self.test_data.LOADER_ASSET_PATH, config_file)
+        lessons_structure = self.test_data.load_yaml_file(complete_config_file_path)
+
+        topic = self.test_data.create_topic("1")
+        unit_plan = self.test_data.create_unit_plan(topic, "1")
+
+        lesson_loader = LessonsLoader(
+            config_file,
+            self.load_log,
+            lessons_structure,
+            topic,
+            unit_plan,
+            self.test_data.LOADER_ASSET_PATH
+        )
+        self.assertRaises(
+            MissingRequiredFieldError,
+            lesson_loader.load,
+        )
+
+    def test_programming_exercises(self):
```

Comment:
Rename `programming_challenges` not `programming_exercises`. Should also have a test for when these are unset.